db/local/db.py

```diff
@@ -0,0 +1,167 @@
+import argparse
+import os
+import sys
+import subprocess
+from typing import Type
+
+from peewee_migrate import Router
+
+from src.base_model import BaseModel, db
+
+MIGRATION_DIRECTORY: str = "db/local/migrations"
+
+models: list[Type[BaseModel]] = BaseModel.__subclasses__()
+
+
+def run(args: argparse.Namespace):
+    """Run the database
+
+    :param args: parsed arguments
+    :type args: argparse.Namespace
+    """
+    name = args.database_name
+    print('Running database...')
+    print(os.getcwd())
+    subprocess.run([
+        "docker-compose", "-f", "db/local/docker-compose.yml", "up",
+        "-d", "-".join(["postgres", name])
+    ])
+
+
+def stop(args: argparse.Namespace):
+    """Stop the database
+
+    :param args: parsed arguments
+    :type args: argparse.Namespace
+    """
+    name = args.database_name
+    print('Stopping database...')
+    subprocess.run([
+        "docker-compose", "-f", "db/local/docker-compose.yml", "stop",
+        "-".join(["postgres", name])
+    ])
+
+
+def create(_: argparse.Namespace):
+    """Create database tables
+    """
+    db.create_tables(models)
+    print('Created tables')
+
+
+def drop(_: argparse.Namespace):
+    """Drop database tables
+    """
+    db.drop_tables(models)
+    print('Dropped tables')
+
+
+def recreate(args: argparse.Namespace):
+    """Drop and create database tables
+
+    :param args: parsed arguments
+    :type args: argparse.Namespace
+    """
+    drop(args)
+    create(args)
+
+
+def fail_on_name(name: str | None):
+    """Fail if no name is provided
+
+    :param name: migration name to test
+    :type name: str | None
+    """
+    if not name:
+        print('Please provide a name for the migration', file=sys.stderr)
+        sys.exit(1)
+
+
+def migration(args: argparse.Namespace):
+    """Create, run or rollback (a) migration(s)
+
+    :param args: parsed arguments
+    """
+    action = args.migration_action
+    name = args.migration_name
+
+    if not os.path.exists(MIGRATION_DIRECTORY):
+        os.mkdir(MIGRATION_DIRECTORY)
+    router = Router(db, migrate_dir=MIGRATION_DIRECTORY)
+
+    if action == 'create':
+        fail_on_name(name)
+        router.create(name)
+        print(f'Created migration: {name}')
+    elif action == 'run':
+        if not name:
+            router.run()
+            print('Ran all unapplied migrations')
+        else:
+            router.run(name)
+            print(f'Ran migration: {name}')
+    elif action == 'rollback':
+        fail_on_name(name)
+        router.rollback(name)
+        print(f'Rolled back migration: {name}')
+
+
+def execute_on_args():
```

Comment:
Please add some documentation with available commands and arguments to this file 😇 This would help using it in the future